service/frontend/workflow_handler_test.go

```diff
@@ -1496,6 +1515,176 @@ func (s *workflowHandlerSuite) TestGetHistory() {
+}
+
+func (s *workflowHandlerSuite) TestGetHistoryReverse() {
+	namespaceID := namespace.ID(uuid.New())
+	namespaceName := namespace.Name("test-namespace")
+	nextEventID := int64(103)
+	lastFirstTxnID := int64(1000)
+	branchToken := []byte{1}
+	we := commonpb.WorkflowExecution{
+		WorkflowId: "wid",
+		RunId:      "rid",
+	}
+	shardID := common.WorkflowIDToHistoryShard(namespaceID.String(), we.WorkflowId, numHistoryShards)
+	req1 := &historyservice.ReadHistoryBranchReverseRequest{
+		NamespaceId:            namespaceID.String(),
+		ShardId:                shardID,
+		BranchToken:            branchToken,
+		MaxEventId:             nextEventID,
+		LastFirstTransactionId: lastFirstTxnID,
+		PageSize:               3,
+		NextPageToken:          []byte{},
+	}
+	s.mockHistoryClient.EXPECT().ReadHistoryBranchReverse(gomock.Any(), req1).Return(&historyservice.ReadHistoryBranchReverseResponse{
+		HistoryEvents: []*historypb.HistoryEvent{
+			{
+				EventId:   int64(100),
+				EventType: enumspb.EVENT_TYPE_WORKFLOW_TASK_COMPLETED,
+			},
+			{
+				EventId:   int64(101),
+				EventType: enumspb.EVENT_TYPE_WORKFLOW_TASK_SCHEDULED,
+			},
+		},
+		NextPageToken: []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9},
+		Size_:         2,
+	}, nil)
+	req2 := &historyservice.ReadHistoryBranchReverseRequest{
+		NamespaceId:            namespaceID.String(),
+		ShardId:                shardID,
+		BranchToken:            branchToken,
+		MaxEventId:             nextEventID,
+		LastFirstTransactionId: lastFirstTxnID,
+		PageSize:               3,
+		NextPageToken:          []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9},
+	}
+	s.mockHistoryClient.EXPECT().ReadHistoryBranchReverse(gomock.Any(), req2).Return(&historyservice.ReadHistoryBranchReverseResponse{
+		HistoryEvents: []*historypb.HistoryEvent{
+			{
+				EventId:   int64(101),
+				EventType: enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_STARTED,
+				Attributes: &historypb.HistoryEvent_WorkflowExecutionStartedEventAttributes{
+					WorkflowExecutionStartedEventAttributes: &historypb.WorkflowExecutionStartedEventAttributes{
+						SearchAttributes: &commonpb.SearchAttributes{
+							IndexedFields: map[string]*commonpb.Payload{
+								"CustomKeywordField":    payload.EncodeString("random-keyword"),
+								"TemporalChangeVersion": payload.EncodeString("random-data"),
+							},
+						},
+					},
+				},
+			},
+		},
+		NextPageToken: []byte{},
+		Size_:         1,
+	}, nil)
+
+	s.mockSearchAttributesProvider.EXPECT().GetSearchAttributes(gomock.Any(), false).Return(searchattribute.TestNameTypeMap, nil)
+	s.mockSearchAttributesMapperProvider.EXPECT().GetMapper(namespaceName).
+		Return(&searchattribute.TestMapper{}, nil).AnyTimes()
+
+	wh := s.getWorkflowHandler(s.newConfig())
+
+	history, token, eventId, err := wh.getHistoryReverse(
+		context.Background(),
+		metrics.NoopMetricsHandler,
+		namespaceID,
+		namespaceName,
+		we,
+		nextEventID,
+		lastFirstTxnID,
+		3,
+		[]byte{},
+		branchToken,
+	)
+	s.NoError(err)
+	s.NotNil(history)
```

Comment:
nit: please move this to just before 1645 where it becomes a necessary precondition

Reply:
I followed suit from the existing `TestGetHistory()` before this test function. If we want to change this, we should change it for the existing tests as well. Should be done in separate PR